components/header.js

```diff
@@ -1,7 +1,39 @@
-import React from 'react';
+"use client";
+import React, { useState } from 'react';
 import pageStyles from '../app/page.module.css';
+import { icons } from '@/components/utilities/card';
 
 const Header = () => {
+  const [headerRules, setHeaderRules] = useState({ cardRank: false, suitRank: false, combinations: false, smashes: false });
+
+  const handleCardRankClick = () => {
+    setHeaderRules({
+      ...headerRules,
+      cardRank: !headerRules.cardRank,
+    });
+  }
+
+  const handleSuitRankClick = () => {
+    setHeaderRules({
+      ...headerRules,
+      suitRank: !headerRules.suitRank,
+    });
+  }
+
+  const handleCombinationsClick = () => {
+    setHeaderRules({
+      ...headerRules,
+      combinations: !headerRules.combinations,
+    })
+  }
+
+  const handleSmashClick = () => {
+    setHeaderRules({
+      ...headerRules,
+      smashes: !headerRules.smashes,
+    })
+  }
```

Comment:
these are all structured the exact same way, and so can be DRYed up a bit as one function. This is untested, but I think something like this would work:

```suggestion
  const handleClick = (type) => {
    setHeaderRules({
      ...headerRules,
     [type]: !headerRules[type],
    });
  }
```

then below you'd just call it like

```html
onClick={handleSmashClick(smashes)}
```

for each of them